contracts/authorities/BondingCurveAuthority.sol

```diff
@@ -0,0 +1,223 @@
+
+        if (
+            amount == 0 || msg.value != bondingCurvePrice + partyDaoFee + treasuryFee + creatorFee
+        ) {
+            revert InvalidMessageValue();
+        }
+
+        partyInfos[party].supply = partyInfo.supply + amount;
+
+        payable(address(party)).transfer(treasuryFee);
+        if (creatorFee != 0) partyInfo.creator.transfer(creatorFee);
+
+        for (uint256 i = 0; i < amount; i++) {
+            party.mint(msg.sender, PARTY_CARD_VOTING_POWER, address(0));
+        }
+    }
+
+    function sellPartyCards(Party party, uint256[] memory tokenIds) external {
+        PartyInfo memory partyInfo = partyInfos[party];
+
+        if (partyInfo.creator == address(0)) {
+            revert PartyNotSupported();
+        }
+
+        uint256 amount = tokenIds.length;
+        uint256 bondingCurvePrice = _getBondingCurvePrice(partyInfo.supply - amount, amount);
+        uint256 partyDaoFee = (bondingCurvePrice * partyDaoFeeBps) / BPS;
+        uint256 treasuryFee = (bondingCurvePrice * treasuryFeeBps) / BPS;
+        uint256 creatorFee = (bondingCurvePrice * partyInfo.creatorFee) / BPS;
+
+        partyInfos[party].supply = uint80(partyInfo.supply - amount);
+
+        for (uint256 i = 0; i < amount; i++) {
+            if (party.ownerOf(tokenIds[i]) != msg.sender) {
+                revert Unauthorized();
+            }
+            party.burn(tokenIds[i]);
+        }
+
+        if (creatorFee != 0) partyInfo.creator.transfer(partyDaoFee);
+        payable(address(party)).transfer(treasuryFee);
+        payable(msg.sender).transfer(bondingCurvePrice - partyDaoFee - treasuryFee - creatorFee);
+    }
+
+    function getPriceToSell(Party party, uint256 amount) external view returns (uint256) {
+        PartyInfo memory partyInfo = partyInfos[party];
+        uint256 bondingCurvePrice = _getBondingCurvePrice(partyInfo.supply - amount, amount);
+        return
+            (bondingCurvePrice * (BPS - partyDaoFeeBps - treasuryFeeBps - partyInfo.creatorFee)) /
+            BPS;
+    }
+
+    function getPriceToBuy(Party party, uint256 amount) external view returns (uint256) {
+        PartyInfo memory partyInfo = partyInfos[party];
+        uint256 bondingCurvePrice = _getBondingCurvePrice(partyInfo.supply, amount);
+        return
+            (bondingCurvePrice * (BPS + partyDaoFeeBps + treasuryFeeBps + partyInfo.creatorFee)) /
+            BPS;
+    }
+
+    function _getBondingCurvePrice(
+        uint256 lowerSupply,
+        uint256 amount
+    ) public pure returns (uint256) {
+        uint256 amountSquared = amount * amount;
+        return
+            (1 ether *
+                (amount *
+                    lowerSupply *
+                    lowerSupply +
+                    (amountSquared - amount) *
+                    lowerSupply +
+                    (2 * amountSquared * amount + amount - 3 * amountSquared) /
+                    6)) /
+            50_000 +
+            amount *
+            0.001 ether;
```

Comment:
Could it be useful to allow changing this `0.001 ether` variable in the future? Or extract this logic to be read from another contract to be more modular? This contract is meant to holds funds for many Parties so it's not easy to redeploy.